Rename some functions related to display refresh

smfr opened the pull request:
#### 6ee1068d110acc9c299bf3e017a3ef192e4157a4
<pre>
Rename some functions related to display refresh
<a href="https://bugs.webkit.org/show_bug.cgi?id=246897">https://bugs.webkit.org/show_bug.cgi?id=246897</a>
rdar://101451193

Reviewed by Ryosuke Niwa.

DrawingArea::didUpdate() -&gt; DrawingArea::displayDidRefresh()
    This is the primary driver of web process &quot;displayDidRefresh&quot; in the RemoteLayerTree
    world; it was called DrawingArea::didUpdate() because, for a long commit, we did call it
    to indicate that the last one has completed. But it&apos;s more common behavior is the
    display refresh signal from the UI process.

RemoteLayerTreeDisplayRefreshMonitor::didUpdateLayers() -&gt; RemoteLayerTreeDisplayRefreshMonitor::triggerDisplayDidRefresh()
    Similar to above, this only meant &quot;did update layers&quot; in the &quot;missed commit&quot; scenario.
    Normally it just means &quot;trigger DisplayRefreshMonitor::displayDidRefresh&quot;.
    I made triggerDisplayDidRefresh() private to make it clear that the friended RemoteLayerTreeDrawingArea
    should be the only caller.

WKOneShotDisplayLinkHandler -&gt; WKDisplayLinkHandler
    It isn&apos;t one-shot.

RemoteLayerTreeDrawingAreaProxy::scheduleDisplayLink/pauseDisplayLink -&gt; RemoteLayerTreeDrawingAreaProxy::scheduleDisplayRefreshCallbacks/pauseDisplayRefreshCallbacks
    On macOS (which has additional needs for a display link in the UI process in addition to just
    notifying the web process) this doesn&apos;t directly correspond with starting/pausing a DisplayLink.

I did not attempt to unify displayDidRefresh/didRefreshDisplay, both of which we use.

* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
(WebKit::DrawingAreaProxyCoordinatedGraphics::update):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
(WebKit::RemoteLayerTreeDrawingAreaProxy::scheduleDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxy::pauseDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxy::scheduleDisplayLink): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxy::pauseDisplayLink): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxy::didRefreshDisplay):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::displayLinkHandler):
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::scheduleDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::pauseDisplayRefreshCallbacks):
(-[WKOneShotDisplayLinkHandler initWithDrawingAreaProxy:]): Deleted.
(-[WKOneShotDisplayLinkHandler dealloc]): Deleted.
(-[WKOneShotDisplayLinkHandler setPreferredFramesPerSecond:]): Deleted.
(-[WKOneShotDisplayLinkHandler displayLinkFired:]): Deleted.
(-[WKOneShotDisplayLinkHandler timerFired]): Deleted.
(-[WKOneShotDisplayLinkHandler invalidate]): Deleted.
(-[WKOneShotDisplayLinkHandler schedule]): Deleted.
(-[WKOneShotDisplayLinkHandler pause]): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::scheduleDisplayLink): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::pauseDisplayLink): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.h:
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::scheduleDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::pauseDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::scheduleDisplayLink): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::pauseDisplayLink): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::displayDidRefresh):
(WebKit::DrawingAreaCoordinatedGraphics::didUpdate): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.h:
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::displayDidRefresh):
(WebKit::DrawingArea::didUpdate): Deleted.
* Source/WebKit/WebProcess/WebPage/DrawingArea.messages.in:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDisplayRefreshMonitor.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDisplayRefreshMonitor.mm:
(WebKit::RemoteLayerTreeDisplayRefreshMonitor::triggerDisplayDidRefresh):
(WebKit::RemoteLayerTreeDisplayRefreshMonitor::didUpdateLayers): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::displayDidRefresh):
(WebKit::RemoteLayerTreeDrawingArea::didUpdate): Deleted.

Canonical link: <a href="https://commits.webkit.org/255889@main">https://commits.webkit.org/255889@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/289c80d40068dc71c7aae343bf49f44a9242e479

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/93845 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3037 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/24416 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/103477 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3053 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31272 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86164 "Built successfully") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/99482 "Hash 289c80d4 for PR 5667 does not build (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99512 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2176 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80268 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29197 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84102 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83807 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72150 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/37668 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/17633 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/35532 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18893 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39411 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/41466 "Found 1 new test failure: http/wpt/service-workers/fetch-service-worker-preload.https.html (failure)") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41346 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38124 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.h:
```diff
@@ -29,7 +29,7 @@
 #if PLATFORM(IOS_FAMILY)
 
-OBJC_CLASS WKOneShotDisplayLinkHandler;
+OBJC_CLASS WKDisplayLinkHandler;
```
Why not WKDisplayRefreshHandler to be consistent?

smfr replied:
Could be, but it wraps a `CADisplayLink`.

rniwa replied:
Hm... I still think WKDisplayRefreshHandler is a more descriptive name, not that I would insist on it.

smfr committed:
Rename some functions related to display refresh
https://bugs.webkit.org/show_bug.cgi?id=246897
rdar://101451193

Reviewed by Ryosuke Niwa.

DrawingArea::didUpdate() -> DrawingArea::displayDidRefresh()
    This is the primary driver of web process "displayDidRefresh" in the RemoteLayerTree
    world; it was called DrawingArea::didUpdate() because, for a long commit, we did call it
    to indicate that the last one has completed. But it's more common behavior is the
    display refresh signal from the UI process.

RemoteLayerTreeDisplayRefreshMonitor::didUpdateLayers() -> RemoteLayerTreeDisplayRefreshMonitor::triggerDisplayDidRefresh()
    Similar to above, this only meant "did update layers" in the "missed commit" scenario.
    Normally it just means "trigger DisplayRefreshMonitor::displayDidRefresh".
    I made triggerDisplayDidRefresh() private to make it clear that the friended RemoteLayerTreeDrawingArea
    should be the only caller.

WKOneShotDisplayLinkHandler -> WKDisplayLinkHandler
    It isn't one-shot.

RemoteLayerTreeDrawingAreaProxy::scheduleDisplayLink/pauseDisplayLink -> RemoteLayerTreeDrawingAreaProxy::scheduleDisplayRefreshCallbacks/pauseDisplayRefreshCallbacks
    On macOS (which has additional needs for a display link in the UI process in addition to just
    notifying the web process) this doesn't directly correspond with starting/pausing a DisplayLink.

I did not attempt to unify displayDidRefresh/didRefreshDisplay, both of which we use.

* Source/WebKit/UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp:
(WebKit::DrawingAreaProxyCoordinatedGraphics::update):
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.h:
(WebKit::RemoteLayerTreeDrawingAreaProxy::scheduleDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxy::pauseDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxy::scheduleDisplayLink): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxy::pauseDisplayLink): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::commitLayerTree):
(WebKit::RemoteLayerTreeDrawingAreaProxy::didRefreshDisplay):
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteLayerTreeDrawingAreaProxyIOS.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::displayLinkHandler):
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::scheduleDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::pauseDisplayRefreshCallbacks):
(-[WKOneShotDisplayLinkHandler initWithDrawingAreaProxy:]): Deleted.
(-[WKOneShotDisplayLinkHandler dealloc]): Deleted.
(-[WKOneShotDisplayLinkHandler setPreferredFramesPerSecond:]): Deleted.
(-[WKOneShotDisplayLinkHandler displayLinkFired:]): Deleted.
(-[WKOneShotDisplayLinkHandler timerFired]): Deleted.
(-[WKOneShotDisplayLinkHandler invalidate]): Deleted.
(-[WKOneShotDisplayLinkHandler schedule]): Deleted.
(-[WKOneShotDisplayLinkHandler pause]): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::scheduleDisplayLink): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxyIOS::pauseDisplayLink): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.h:
* Source/WebKit/UIProcess/RemoteLayerTree/mac/RemoteLayerTreeDrawingAreaProxyMac.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::scheduleDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::pauseDisplayRefreshCallbacks):
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::scheduleDisplayLink): Deleted.
(WebKit::RemoteLayerTreeDrawingAreaProxyMac::pauseDisplayLink): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::displayDidRefresh):
(WebKit::DrawingAreaCoordinatedGraphics::didUpdate): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.h:
* Source/WebKit/WebProcess/WebPage/DrawingArea.h:
(WebKit::DrawingArea::displayDidRefresh):
(WebKit::DrawingArea::didUpdate): Deleted.
* Source/WebKit/WebProcess/WebPage/DrawingArea.messages.in:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDisplayRefreshMonitor.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDisplayRefreshMonitor.mm:
(WebKit::RemoteLayerTreeDisplayRefreshMonitor::triggerDisplayDidRefresh):
(WebKit::RemoteLayerTreeDisplayRefreshMonitor::didUpdateLayers): Deleted.
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.h:
* Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm:
(WebKit::RemoteLayerTreeDrawingArea::displayDidRefresh):
(WebKit::RemoteLayerTreeDrawingArea::didUpdate): Deleted.

Canonical link: https://commits.webkit.org/255889@main
